pkgs/development/tools/build-managers/bazel/default.nix

```diff
@@ -56,9 +56,396 @@ stdenv.mkDerivation rec {
+      unfiltered_cxx_flag: "-no-canonical-prefixes"
+      unfiltered_cxx_flag: "-fno-canonical-system-headers"
+
+      # Make C++ compilation deterministic. Use linkstamping instead of these
+      # compiler symbols.
+      unfiltered_cxx_flag: "-Wno-builtin-macro-redefined"
+      unfiltered_cxx_flag: "-D__DATE__=\"redacted\""
+      unfiltered_cxx_flag: "-D__TIMESTAMP__=\"redacted\""
+      unfiltered_cxx_flag: "-D__TIME__=\"redacted\""
+
+      # Security hardening on by default.
+      # Conservative choice; -D_FORTIFY_SOURCE=2 may be unsafe in some cases.
+      # We need to undef it before redefining it as some distributions now have
+      # it enabled by default.
+      compiler_flag: "-U_FORTIFY_SOURCE"
+      compiler_flag: "-D_FORTIFY_SOURCE=1"
+      compiler_flag: "-fstack-protector"
+      linker_flag: "-Wl,-z,relro,-z,now"
+
+      # Enable coloring even if there's no attached terminal. Bazel removes the
+      # escape sequences if --nocolor is specified. This isn't supported by gcc
+      # on Ubuntu 14.04.
+      # compiler_flag: "-fcolor-diagnostics"
+
+      # All warnings are enabled. Maybe enable -Werror as well?
+      compiler_flag: "-Wall"
+      # Enable a few more warnings that aren't part of -Wall.
+      compiler_flag: "-Wunused-but-set-parameter"
+      # But disable some that are problematic.
+      compiler_flag: "-Wno-free-nonheap-object" # has false positives
+
+      # Keep stack frames for debugging, even in opt mode.
+      compiler_flag: "-fno-omit-frame-pointer"
+
+      # Anticipated future default.
+      linker_flag: "-no-canonical-prefixes"
+      # Have gcc return the exit code from ld.
+      linker_flag: "-pass-exit-codes"
+      # Gold linker only? Can we enable this by default?
+      # linker_flag: "-Wl,--warn-execstack"
+      # linker_flag: "-Wl,--detect-odr-violations"
+
+      compilation_mode_flags {
+        mode: DBG
+        # Enable debug symbols.
+        compiler_flag: "-g"
+      }
+      compilation_mode_flags {
+        mode: OPT
+
+        # No debug symbols.
+        # Maybe we should enable https://gcc.gnu.org/wiki/DebugFission for opt or
+        # even generally? However, that can't happen here, as it requires special
+        # handling in Bazel.
+        compiler_flag: "-g0"
+
+        # Conservative choice for -O
+        # -O3 can increase binary size and even slow down the resulting binaries.
+        # Profile first and / or use FDO if you need better performance than this.
+        compiler_flag: "-O2"
+
+        # Disable assertions
+        compiler_flag: "-DNDEBUG"
+
+        # Removal of unused code and data at link time (can this increase binary size in some cases?).
+        compiler_flag: "-ffunction-sections"
+        compiler_flag: "-fdata-sections"
+        linker_flag: "-Wl,--gc-sections"
+      }
+      linking_mode_flags { mode: DYNAMIC }
+    }
+  '');
+
+  osxCcWrapperFile = writeScript "osx_cc_wrapper.sh" (if stdenv.isDarwin then ''
```

Comment:
We can’t seriously copy files verbatim into our package expressions, with legal headers even. It should be put into a (temporary if possible, well documented) file instead.